spec/unit/ec2_server_create_spec.rb

```diff
@@ -2113,5 +2112,39 @@
         end
       end
     end
+
+    describe "--security-group-ids option" do
+      before do
+        allow(Fog::Compute::AWS).to receive(:new).and_return(@ec2_connection)
+      end
+
+      context "when invalid input is given for --security-group-ids from cli" do
+        it "raises error" do
+          @knife_ec2_create.config[:security_group_ids] = 'sg-aabbccdd; sg-3764sdss ; sg-00aa11bb'
+          expect { @knife_ec2_create.validate! }.to raise_error(SystemExit)
+        end
+      end
+
+      context "when valid input is given --security-group-ids from cli" do
+        it "does not raise error" do
+          @knife_ec2_create.config[:security_group_ids] = 'sg-aabbccdd, sg-3764sdss, sg-00aa11bb'
+          expect { @knife_ec2_create.validate! }.to_not raise_error
+        end
+      end
+
+      context "when invalid input is given for --security-group-ids from knife.rb" do
```

Comment:
We should still allow a comma separated list in a string in the configuration file, and use the same code to parse it that we use when we take a comma separated list from the command line.

We should also allow an array in the configuration file. It is ruby, so the user is likely to try to use the array. We allow arrays in other places in the configuration files, like `cookbook_path` and `ohai.disabled_plugins`. We don't have to document it in the user documentation (although it wouldn't hurt). I'd expect a test like `it "allows --security-group-ids set from an Array in knife.rb"`.